### PR TITLE
Use superseded and deprecated configuration from pillar

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Updating the superseded and deprecated decorators to work when specified as pillar values.

### What issues does this PR fix or reference?
#42514 

### Previous Behavior
Previously the list of functions for use_superseded and use_deprecated were specified in minion configuration.

### New Behavior
Specify the list of functions for use_superseded and use_deprecated as pillar values.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

File: salt/utils/decorators/__init__.py [hunk 1]
```diff
@@ -542,8 +542,14 @@ def _set_function(self, function):
...
         opts = self._globals.get('__opts__', '{}')
-        use_deprecated = full_name in opts.get(self.CFG_USE_DEPRECATED, list())
-        use_superseded = full_name in opts.get(self.CFG_USE_SUPERSEDED, list())
+        pillar = self._globals.get('__pillar__', '{}')
+
+        use_deprecated = full_name in opts.get(self.CFG_USE_DEPRECATED, list()) or \
+            full_name in pillar.get(self.CFG_USE_DEPRECATED, list())
+
+        use_superseded = full_name in opts.get(self.CFG_USE_SUPERSEDED, list()) or \
+            full_name in pillar.get(self.CFG_USE_SUPERSEDED, list())
```
Comment (isbm) [inline salt/utils/decorators/__init__.py]: @garethgreenaway Please do me a favour: remove the `\` thing enclosing whole expression in `(...)` :-)

**Review by isbm — APPROVED** [review]
@garethgreenaway :+1: for this and + :100: and one hug for great unit tests! Great job!

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Pull request merged**